### PR TITLE
feat: macOS removal fallback when moveItemToTrash fails

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Refs https://github.com/electron/electron/issues/4974.

Some macOS volumes may not support a Trash folder or it may be disabled, so these methods will report failure by returning NO or `nil` and an `NSError` with `NSFeatureUnsupportedError`.

This enables consumers to handle this by optionally passing a boolean `deleteOnFail` value to indicate whether or not the item should be unilaterally deleted if the `NSFileManager` fails to execute `trashItemAtURL`.

cc @miniak @ckerr @MarshallOfSound 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Enable macOS users to fallback to item removal when when `shell.moveItemToTrash` fails.

### CODE REVIEW

File: shell/common/platform_util.h [hunk 1]
```diff
@@ -41,7 +42,7 @@ void OpenExternal(const GURL& url,
                   OpenExternalCallback callback);
 
 // Move a file to trash.
-bool MoveItemToTrash(const base::FilePath& full_path);
+bool MoveItemToTrash(const base::FilePath& full_path, mate::Arguments* args);
```
Comment (deepak1556) [inline shell/common/platform_util.h]: The function signature can be just `bool MoveItemToTrash(mate::Arguments* args)` and the `const base::FilePath& full_path` value can be still be checked for non-optionality with `args->Length()`.

Comment (MarshallOfSound) [reply shell/common/platform_util.h]: Putting the filrpath in the signature though ensures the error message for missing arguments is consistent across methods 

#justmy2cents

Comment (deepak1556) [reply shell/common/platform_util.h]: In that case the implementation declaration should not be made aware of `mate::Arguments`, I would suggest doing the runtime parsing in `shell/common/api/*.cc` and call the implementation methods with right values.

File: shell/common/api/atom_api_shell.cc [hunk 2]
```diff
@@ -71,6 +71,13 @@ v8::Local<v8::Promise> OpenExternal(const GURL& url, mate::Arguments* args) {
   return handle;
 }
 
+bool MoveItemToTrash(const base::FilePath& full_path, mate::Arguments* args) {
```
Comment (deepak1556) [inline shell/common/api/atom_api_shell.cc]: `bool MoveItemToTrash(mate::Arguments* args)` , try to reuse args and perform the required processing

**Pull request merged**